nsqd/options.go

```diff
@@ -75,61 +71,18 @@ type Options struct {
 }
 
 func NewOptions() *Options {
-	hostname, err := os.Hostname()
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	h := md5.New()
-	io.WriteString(h, hostname)
-	defaultID := int64(crc32.ChecksumIEEE(h.Sum(nil)) % 1024)
-
 	return &Options{
-		ID: defaultID,
-
-		TCPAddress:       "0.0.0.0:4150",
-		HTTPAddress:      "0.0.0.0:4151",
-		HTTPSAddress:     "0.0.0.0:4152",
-		BroadcastAddress: hostname,
-
 		NSQLookupdTCPAddresses: make([]string, 0),
 		AuthHTTPAddresses:      make([]string, 0),
 
-		MemQueueSize:    10000,
-		MaxBytesPerFile: 100 * 1024 * 1024,
-		SyncEvery:       2500,
-		SyncTimeout:     2 * time.Second,
-
 		QueueScanInterval:        100 * time.Millisecond,
 		QueueScanRefreshInterval: 5 * time.Second,
 		QueueScanSelectionCount:  20,
 		QueueScanWorkerPoolMax:   4,
 		QueueScanDirtyPercent:    0.25,
 
-		MsgTimeout:    60 * time.Second,
-		MaxMsgTimeout: 15 * time.Minute,
-		MaxMsgSize:    1024 * 1024,
-		MaxBodySize:   5 * 1024 * 1024,
-		MaxReqTimeout: 1 * time.Hour,
 		ClientTimeout: 60 * time.Second,
 
-		MaxHeartbeatInterval:   60 * time.Second,
```

Comment:
Thinking on this a little more I think it'd be better to preserve the defaults here in `NewOptions()` and just change the order in `main()`  to get `NewOptions()` first and pass that in to `nsqFlagset()`, so that the defaults come from the `nsqd` package, not `apps/nsqd`